### PR TITLE
feat(template): make rxFor stable

### PR DESCRIPTION
**Pull request by hoebbelsB**
# Description

this PR makes sure `rxFor` is in a state where it can be considered as stable and moved from the experimental package.
I've implemented a test suite which basically resembles the original [ngFor spec](https://github.com/angular/angular/blob/main/packages/common/test/directives/ng_for_spec.ts).

* [x] move `rxFor` to `template/`
* [x] implement migration script 
* [x] make sure `rxFor` acts as drop-in replacement for `ngFor`
* [x] implement unit tests covering the `ngFor` behavior
* [x] implement unit tests covering `rxFor` specific API

### CODE REVIEW

**Review by edbzn — COMMENTED** [review]
Could explain to me the idea of tracking by "context" where you pass the component instance reference to the `trackBy` fn, I'm not sure to understand the idea. 

It would be nice to provide an SSR test to be sure rxFor is rendering on the server.

File: libs/template/experimental/for/src/lib/tests/fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const TEMPLATE =
+  '<div><span *rxFor="let item of items">{{item.toString()}};</span></div>';
+
+export function createTestComponent(
+  template: string = TEMPLATE
+): ComponentFixture<TestComponent> {
+  return TestBed.overrideComponent(TestComponent, {
+    set: { template: template },
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/template/experimental/for/src/lib/tests/fixtures.ts:41]: this way each `it` block can create it's very own template :). I've copied it from https://github.com/angular/angular/blob/main/packages/common/test/directives/ng_for_spec.ts

File: libs/template/experimental/for/src/lib/tests/for.directive.strategy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    ['idle'],
+    ['native'],
+  ])('Strategy: %p', (strategy) => {
+    it('should render with given strategy', (done) => {
+      component.strategy = strategy;
+      component.renderedValue$.subscribe((v) => {
+        expect(v).toEqual([1, 2]);
+        expect(nativeElement.textContent).toBe('1;2;');
+        done();
+      });
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/template/experimental/for/src/lib/tests/for.directive.strategy.spec.ts:55]: but as you can see, the dom is tested? `nativeElement.textContent`

File: libs/template/experimental/for/src/lib/tests/for.directive.strategy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      fixture.detectChanges();
+    });
+    it('should not affect primary strategy', (done) => {
+      component.strategy = strategy;
+      component.renderedValue$.subscribe((v) => {
+        expect(v).toEqual([1, 2]);
+        expect(nativeElement.textContent).toBe('1;2;');
+        expect(strategyProvider.primaryStrategy).toBe(primaryStrategy);
+        done();
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/template/experimental/for/src/lib/tests/for.directive.strategy.spec.ts:64]: same here

File: libs/template/experimental/for/src/lib/tests/fixtures.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }).createComponent(TestComponent);
+}
+
+export function createErrorHandler() {
```
[the remark this answers is not in the document]
Comment (hoebbelsB) [reply libs/template/experimental/for/src/lib/tests/fixtures.ts:45]: this way it is consistent with `createComponent`

File: libs/template/experimental/for/src/lib/tests/fixtures.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (edbzn) [reply libs/template/experimental/for/src/lib/tests/fixtures.ts:41]: Good to know this trick!

File: apps/ssr-e2e/src/integration/app.spec.ts [hunk 5]
```diff
@@ -20,4 +20,18 @@ describe('@rx-angular/template universal rendering', () => {
       )
     });
   });
+
+  describe('RxFor', () => {
+    it('should display green and purple text', () => {
+      cy.request('http://localhost:4200').its('body').then(
+        html => {
+          console.log(html)
```
Comment (niklaas) [inline apps/ssr-e2e/src/integration/app.spec.ts:28]: Not sure what's your process is for this repo but you might want to remove this.

**Review by niklaas — COMMENTED** [review]
Nice to see this going to production! 🙌🏼 

Just took a very quick look and saw the following:

**Pull request merged**